digitalearthau/stacker.py

```diff
@@ -0,0 +1,242 @@
+"""
+A DEA version of `stacker` from ODC
+
+Uses the exact same implementation, but wraps it in QSUB Running/Event logging goodness.
+
+As such, it is a three step process, with the second two happening automatically after the first:
+
+1. Submit
+
+ Submit a single threaded PBS job to calculate how big the requested job is.
+ This step executes very quickly, creates the 'job' directory which groups log files
+ and metadata for the full job, then finally schedules a 'generate' job to run using
+ PBS qsub.
+
+2. Generate
+
+ Run as a single threaded process inside PBS to:
+ calculate how much work is required,
+ write the tasks out to a task file and then
+ schedule another PBS job of an appropriate size to 'Run' the stacking
+
+3. Run
+
+ Do all of the stacking. Spin up a Redis queue and use celery to farm the tasks
+ out across a large PBS job.
+"""
+import logging
+from datetime import datetime
+from functools import partial
+from math import ceil
+from pathlib import Path
+from typing import Tuple
+
+import click
+
+import datacube
+import digitalearthau
+from datacube.api.query import Query
+from datacube.index._api import Index
+from datacube.ui import click as ui
+from datacube.ui import task_app
+from datacube_apps.stacker import stacker
+from digitalearthau import __version__
+from digitalearthau import paths, serialise
+# pylint: disable=invalid-name
+from digitalearthau.qsub import with_qsub_runner, TaskRunner
+from digitalearthau.runners.model import TaskDescription
+from digitalearthau.runners.util import init_task_app, submit_subjob
+
+_LOG = logging.getLogger(__file__)
+APP_NAME = 'dea-stacker'
+
+
+@click.group(help='DEA Stacker\n\n' + __doc__)
```

Comment:
I never thought of using `__doc__` to deduplicate click help. I'm going to copy this in the other scripts :)

Reply:
Thanks.
I'd never thought of it earlier either but it works really well. It's great having lots of information available on the command line.